### PR TITLE
Disable Some Einsum ORTModule Tests Due to Issue from PyTorch Exporter

### PR DESCRIPTION
**Pull request by iK1D**
In PyTorch 1.11.0, there is issue in PyTorch exporter that fail to handle reduce node shape, so if there is ReduceProd in the sub-graph of Einsum, ORTModule will fail to run. Disable such tests for now. Will enable them once the issue in PyTorch is fixed.

This PR also contains a bug fix to Reshape node when generating Einsum sub-graph.

### CODE REVIEW

File: orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py [hunk 1]
```diff
@@ -1183,89 +1187,89 @@ def run_step(model, input_left, input_right):
...
-            pt_input_left = torch.rand(lhs_shape, device=device)
-            pt_input_right = torch.rand(rhs_shape, device=device)
-            ort_input_left = copy.deepcopy(pt_input_left)
-            ort_input_right = copy.deepcopy(pt_input_right)
-            pt_prediction = run_step(pt_model, pt_input_left, pt_input_right)
-            ort_prediction = run_step(ort_model, ort_input_left, ort_input_right)
-
-            _test_helpers.assert_values_are_close(ort_prediction, pt_prediction, atol=1e-3, rtol=1e-3)
-            _test_helpers.assert_gradients_match_and_reset_gradient(ort_model, pt_model, atol=1e-3, rtol=1e-3)
+# def test_gradient_correctness_einsum_2():
```
Comment (xadupre) [inline orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py:1190]: skipping it would be better than commenting it out.

Comment (iK1D) [reply orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py]: Right! Actually I can disable them for specific torch version only. Thanks!

File: orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py [hunk 2]
```diff
@@ -1184,6 +1190,11 @@ def run_step(model, input_left, input_right):
         _test_helpers.assert_gradients_match_and_reset_gradient(ort_model, pt_model, atol=1e-3, rtol=1e-3)
 
 def test_gradient_correctness_einsum_2():
+    # In PyTorch 1.11.0, there is issue during reduce node shape handling for exporter that
```
Comment (baijumeswani) [inline orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py:1193]: shall we use `@pytest.mark.skipif`? And add a TODO so we remove it in the future.

File: orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py [hunk 3]
```diff
@@ -1134,8 +1134,14 @@ def run_step(model, input):
...
-@pytest.mark.parametrize("equation", ["s,se->se", "se,sc->sec", "se,se->s", "sec,sm->ecm",
-                                      "sec,ecm->sm", "ks,ksm->sm", "kes,ems->mek", "kes,ksm->ms"])
+# In PyTorch 1.11.0, there is issue during reduce node shape handling for exporter, so any sub-graph that
+# contains ReduceProd will fail to run, for example, "sec,sm->ecm", "sec,ecm->sm".
+# Currently move these cases out from the tests, will enable these tests again once the issue in PyTorch is fixed.
+einsum_equations = ["s,se->se", "se,sc->sec", "se,se->s", "ks,ksm->sm", "kes,ems->mek", "kes,ksm->ms"]
+if LooseVersion(torch.__version__) < LooseVersion('1.11.0'):
+    einsum_equations.append("sec,sm->ecm")
+    einsum_equations.append("sec,ecm->sm")
+@pytest.mark.parametrize("equation", einsum_equations)
```
Comment (baijumeswani) [inline orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py:1144]: I think this is fine, but you could consider using `pytest.param("sec,sm->ecm", marks=pytest.mark.skipif(LooseVersion(torch.__version__) >= LooseVersion('1.11.0'), reason="torch 1.11.0 incompatible")),`.
Let's add a TODO here too.

**Review by xadupre — APPROVED** [review]

**Comment by baijumeswani** [comment]
Thanks for making this change.

**Pull request merged**